commit 28e893e56d87f6d8266174ce9c4bc98b03c0e953
Author: Moritz "WanzenBug" Wanzenböck <moritz@wanzenbug.xyz>
Date:   Wed Feb 19 15:56:07 2025 +0100

    secretservice: fix null derefence on locked collections
    
    secret_item_get_secret() may return null if an item is locked or not loaded.
    While we set SECRET_SEARCH_LOAD_SECRETS and SECRET_SEARCH_UNLOCK, there may
    still be locked items, for example the user may refuse the unlock request.
    
    So we still need to check if the secret data is NULL before we can try to
    reference it.
    
    Signed-off-by: Moritz "WanzenBug" Wanzenböck <moritz@wanzenbug.xyz>

diff --git a/secretservice/secretservice.c b/secretservice/secretservice.c
index b77a0d2..938bf9a 100644
--- a/secretservice/secretservice.c
+++ b/secretservice/secretservice.c
@@ -83,6 +83,9 @@ GError *get(char *server, char **username, char **secret) {
 				}
 				g_free(value);
 				secretValue = secret_item_get_secret(l->data);
+				if (secretValue == NULL) {
+					continue;
+				}
 				if (secret != NULL) {
 					*secret = strdup(secret_value_get(secretValue, &length));
 					secret_value_unref(secretValue);
